code/game/objects/items/stacks/medical.dm

```diff
@@ -43,15 +43,20 @@
 		H.UpdateDamageIcon()
 
 	else
+		if (!M.bruteloss && !M.fireloss)
+			user << "<span class='notice'> [M] seems healthy, there are no wounds to treat! </span>"
+			return 1
 
-		M.heal_organ_damage((src.heal_brute/2), (src.heal_burn/2))
-		user.visible_message(
-			SPAN_NOTICE("[M] has been applied with [src] by [user]."),
-			SPAN_NOTICE("You apply \the [src] to [M].")
-		)
-		if(prob(10 + user.stats.getStat(STAT_BIO)))
-			user << SPAN_NOTICE("You have managed to waste less [src].")
-		else
+		user.visible_message( \
+				SPAN_NOTICE("[user] starts applying [src] to [M]."), \
+				SPAN_NOTICE("You start applying [src] to [M].") \
+			)
+		if (do_after(user, 30, M))
```

Comment:
you seem to have overwritten some stat interactions here. Put some back in
could be more creative like changing the healing effectiveness